app/services/authorizer.rb

```diff
@@ -173,4 +173,14 @@ def base_ids
 
     @base_ids ||= (@base_collection.all? { |i| i.is_a?(Integer) }) ? @base_collection : @base_collection.map(&:id)
   end
+
+  def existing_permissions
+    Rails.cache.fetch(cache_key) do
+      user.permissions.select(:name, :resource_type)
+    end
+  end
+
+  def cache_key
+    { :"user_#{user.id}" => "#{user.last_login_on.to_i}-#{user.updated_at.to_i}" }
```

Comment:
Does this mean that user needs to relogin after admin changes his/her permissions to drop the cache?